app/src/main/java/com/github/sdpteam15/polyevents/model/database/remote/FirestoreDatabaseProvider.kt

```diff
@@ -123,45 +126,53 @@ object FirestoreDatabaseProvider : DatabaseInterface {
         }
         firestore!!
             .collection(collection.value)
-            .add(adapter.toDocument(element))
+            .add(adapter.toDocument(element)!!)
             .addOnSuccessListener(successListener)
             .addOnFailureListener(failureListener)
         return ended
     }
 
-    override fun <T : Any> addEntity(
-        element: T,
-        collection: DatabaseConstant.CollectionConstant,
-        adapter: AdapterToDocumentInterface<in T>
-    ): Observable<Boolean> =
-        addEntityAndGetId(element, collection, adapter).mapOnce { it != "" }.then
-
     override fun <T : Any> addListEntity(
         elements: List<T>,
         collection: DatabaseConstant.CollectionConstant,
         adapter: AdapterToDocumentInterface<in T>
-    ): Observable<Pair<Boolean, List<String>?>> {
+    ): Observable<Pair<Boolean, List<String?>>> {
         if (elements.isEmpty())
             return Observable(Pair(true, listOf()))
-        val ended = Observable<Pair<Boolean, List<String>?>>()
-        val mutableList = mutableListOf<String?>()
+        val ended = Observable<Pair<Boolean, List<String?>>>()
+        val mutableList = mutableListOf<Pair<Boolean, String?>?>()
         for (elementWithIndex in elements.withIndex()) {
             mutableList.add(null)
             firestore!!
                 .collection(collection.value)
-                .add(adapter.toDocument(elementWithIndex.value))
+                .add(adapter.toDocument(elementWithIndex.value)!!)
                 .addOnSuccessListener {
                     synchronized(this) {
```

Comment:
You should really consider to comment all of this code in your generic implementation. It's really really hard to understand even if wrote queries during the whole project. By having class like this one, we risk to be penalised for the grading for code complexity. Also, people that will bring back our code will have a really hard time to understand

Reply:
Yes, will be don in my next PR